### DISCUSSION TITLE
Tabs on Startup

### DISCUSSION THREAD

**Discussion by leslieamartinez** [kovidgoyal/kitty#4125] [Q&A] [answered]
I would like to open two tabs. First tab should be the active tab after start up.

1. Tab - Normal terminal session - Labelled xxxx
2. Tab - ssh user@xxx.xxx.xxx -t tmux a (ssh to existing tmux session) - Labelled yyyy

I read through the documentation and it seem simple enough but it just doesn't work. Kitty.conf has "startup_session ~/.config/kitty/kitty-start". Below is my kitty-start

new_tab xxxx
focus
new_tab yyyy
launch ssh user@xxx.xxx.xxx -t tmux a

Your help is greatly appreciated.

**Comment by page-down** [comment, marked as answer]
The first tab is created by default and there is no need to call `new_tab`.
Each tab should have a window created using the `launch` command.
If you do not use `focus` to explicitly specify the final active window inside the corresponding tab, then the first tab is active by default.

Option 1) Use the implicit default focus, on first tab.

```text
# first tab
launch --title first

# second tab
new_tab
launch --title second
```

Option 2) Explicitly defines the window to be activated.

```text
# first tab
launch --title first
# focus on this window in first tab
focus

# second tab
new_tab
launch --title second
```

Option 3) You can use the `--keep-focus` (later versions will have the alias `--dont-take-focus`) option when calling the launch command, which means that the window will not be activated.

```text
# first tab
# window in tab-1
launch --title first

# second tab
new_tab
# window in tab-2 without switch focus
launch --title second --keep-focus
```

**Comment by page-down** [comment]
@kovidgoyal 
I gave it a second thought and it seemed a little odd.

For the remote control `launch` command, the **subjects** of the options all point to the newly opened window or tab. (In terms of natural language.)
For example, `--title`, `--location`, `--cwd`, `--no-response`, etc., all refer to this new window.
However, the subject (target) of the `--keep-focus` option has changed to refer to the "currently active window" keeping the focus.

Like `--no-response`, it seems to be syntactically smoother to use `--no-focus`. This means that the window created with this command will not get the focus. May be considered as an alias for backward compatibility.

```text
kitty @ launch --title new-window --no-focus
```

These are my personal opinions from the user's point of view. Does this make sense to you? Thank you.

**Comment by kovidgoyal** [comment]
Sure I have added --dont-take-focus as an alias

**Comment by leslieamartinez** [comment]
Thanks for your assistance. I feel like I am only scratching the surface of what this terminal can do. I have looked through the docs but I don't know what I don't know. Is there anywhere where you can see what others have been doing with Kitty. Thanks

BTW How would I make the Kitty open the right of the screen?

**Reply by page-down** [reply]
You can start with the most basic functions.
https://sw.kovidgoyal.net/kitty/overview/

Check the documentation when you need to use a certain feature. Also search GitHub to see if there are previous issues mentioned. You don't need to know all the features in one shot.

If you need to adjust the position of the initialization window, you need window manager under Linux to do so, or third-party software or automation scripts under macOS.

**Reply by leslieamartinez** [reply]
Thanks again will do.
